pkg/proxy/backend/backend_conn_mgr.go

```diff
@@ -58,6 +66,13 @@ type redirectResult struct {
 	to   string
 }
 
+const (
+	statusConnected = 1 << iota
```

Comment:
Isn't connected and closing exclusive? We can not just append new flags, also need to clear old flags.

Maybe we should consider exclusive status first.